pkg/virt-api/webhooks/validating-webhook/admitters/vmexport-admitter.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package admitters
+
+import (
+	"encoding/json"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	exportv1 "kubevirt.io/api/export/v1alpha1"
+	webhookutils "kubevirt.io/kubevirt/pkg/util/webhooks"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+// VMExportAdmitter validates VirtualMachineExports
+type VMExportAdmitter struct {
+	Config *virtconfig.ClusterConfig
+}
+
+// NewVMExportAdmitter creates a VMExportAdmitter
+func NewVMExportAdmitter(config *virtconfig.ClusterConfig) *VMExportAdmitter {
+	return &VMExportAdmitter{
+		Config: config,
+	}
+}
+
+// Admit validates an AdmissionReview
+func (admitter *VMExportAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	if ar.Request.Resource.Group != exportv1.SchemeGroupVersion.Group ||
+		ar.Request.Resource.Resource != "virtualmachineexports" {
+		return webhookutils.ToAdmissionResponseError(fmt.Errorf("unexpected resource %+v", ar.Request.Resource))
+	}
+
+	if ar.Request.Operation == admissionv1.Create && !admitter.Config.VMExportEnabled() {
+		return webhookutils.ToAdmissionResponseError(fmt.Errorf("vm export feature gate not enabled"))
+	}
+
+	vmExport := &exportv1.VirtualMachineExport{}
+	// TODO ideally use UniversalDeserializer here
+	err := json.Unmarshal(ar.Request.Object.Raw, vmExport)
+	if err != nil {
+		return webhookutils.ToAdmissionResponseError(err)
+	}
+
+	var causes []metav1.StatusCause
+
+	switch ar.Request.Operation {
+	case admissionv1.Create:
+		sourceField := k8sfield.NewPath("spec", "source")
+
+		if vmExport.Spec.Source.APIGroup == nil {
```

Comment:
I think it is maybe okay to have this as nil, at least with PVC source

Reply:
Yeah, removed that check completely. Especially since we only support PVCs for now, no need to make it more complicated.